dolphin/api/v1/storages.py

```diff
@@ -110,6 +110,14 @@ def create(self, req, body):
                 exception.StorageNotFound) as e:
             raise exc.HTTPBadRequest(explanation=e.msg)
 
+        # Registration success, sync resource collection for this storage
+        try:
+            self.sync(req, storage.id)
+        except Exception as e:
+            # Unexpected error occurred while syncing resources.
+            msg = _('Failed to sync resources for storage: %(storage)s. '
+                    'Error: %(err)s') % {'storage': storage.id, 'err': e}
```

Comment:
If sync fails should there be a corresponding action? Or is it ok to log and go ahead?

Reply:
If sync fails, suggest to trigger it again after a specific time, and if fails too much time, need report alert .

Reply:
Thanks, As discussed I have opened an issue to have model and interface for internal Alert mechanism. https://github.com/sodafoundation/SIM/issues/130